src/api/app/components/workflow_run_detail_component.rb

```diff
@@ -24,14 +24,25 @@ def parse_payload(workflow_run)
     workflow_run.request_payload
   end
 
-  # Old workflow run entries didn't store the configuration-related information
-  def workflow_configuration_data
-    return content_tag(:p, 'This information is not available.') unless workflow_run.configuration_source
-
-    content_tag(:h5, "Workflow Configuration File #{workflow_run.workflow_configuration_url.present? ? 'URL' : 'Path'}").concat(
-      content_tag(:pre,
-                  workflow_run.configuration_source,
-                  class: 'border p-2')
-    )
+  def workflow_configuration_content
+    # get the parsed slice of interest
+    workflow_config_output = parse_workflow_configuration_yaml(workflow_run.workflow_configuration)
+
+    # if the yaml sub-content was parsed and found properly
+    workflow_config_output.join("\n") if workflow_config_output.present?
+  end
+
+  # flag for workflow configuration data availability
+  def configuration_source_not_available
+    [workflow_run.workflow_configuration, workflow_run.workflow_configuration_url, workflow_run.workflow_configuration_path].all?(&:blank?)
+  end
+
+  def parse_workflow_configuration_yaml(blob_content)
+    return if blob_content.blank?
+
+    # parse the workflow_configuration file content previously stored
+    yaml_parsed = YAML.safe_load(blob_content)
+    # extract only the slice of interest
+    yaml_parsed.dig('payload', 'blob', 'rawLines') if yaml_parsed.is_a?(Hash)
```

Comment:
In my tests, this step is not necessary, there is no `payload -> blob -> rawlines` nodes if it's a Hash...

Reply:
Using the value as it is (WorkflowRun#workflow_configuration) in the view, it's displayed correctly.
I don't know why parsing is needed.

Reply:
I also get nil with `payload -> blob -> rawlines`.

I'll go without parsing until we have a good reason for that.

Reply:
Interesting, in my live test (against GitHub for instance), I was receiving the whole structure with that tree and I had to extract the part I needed. 🤔 😕 
Anyway, if the two of you had different output, the majority is set. 👍

Reply:
I picked up 10+ random samples from production and all of them look quite "normal":

```
"---\npr:\n  steps:\n  - branch_package:\n      source_project: [...]
```
```
"version: '1.1'\ntestbuild:\n  steps:\n    - branch_package:\n        source_project: [...]
```
```
"factory-rebuild:\n  steps:\n    - trigger_services:\n       [...]
```